apollo-router/src/registry/mod.rs

```diff
@@ -0,0 +1,175 @@
+use std::string::FromUtf8Error;
+
+use docker_credential::CredentialRetrievalError;
+use docker_credential::DockerCredential;
+use oci_client::Client as ociClient;
+use oci_client::Reference;
+use oci_client::errors::OciDistributionError;
+use oci_client::secrets::RegistryAuth;
+use thiserror::Error;
+
+/// Configuration for fetching an OCI Bundle
+/// This struct does not change on router reloads - they are all sourced from CLI options.
+#[derive(Debug, Clone)]
+pub struct OCIConfig {
+    /// The Apollo key: `<YOUR_GRAPH_API_KEY>`
+    pub apollo_key: String,
+
+    // Should this be a reference instead?
```

Comment:
Generally, it's best to have a struct own its data. Use a reference when these factors apply:
 - performance/resources are a concern
 - a different struct (with a predictable lifetime) clearly owns the data
 - consistency across entities is an important aspect of the problem